src/event.js

```diff
@@ -428,75 +398,40 @@ jQuery.event = {
 		return handlerQueue;
 	},
 
-	// Includes all common event props including KeyEvent and MouseEvent specific props
-	props: ( "altKey bubbles cancelable ctrlKey detail eventPhase " +
-		"metaKey shiftKey view which char charCode key keyCode " +
-		"button buttons clientX clientY offsetX offsetY pageX pageY screenX screenY toElement"
-	).split( " " ),
-
-	propHooks: {
-		which: function( event ) {
-			var button = event.button;
-
-			// Add which for key events
-			if ( event.which == null && rkeyEvent.test( event.type ) ) {
-				return event.charCode != null ? event.charCode : event.keyCode;
-			}
-
-			// Add which for click: 1 === left; 2 === middle; 3 === right
-			if ( !event.which && button !== undefined && rmouseEvent.test( event.type ) ) {
-				return ( button & 1 ? 1 : ( button & 2 ? 3 : ( button & 4 ? 2 : 0 ) ) );
-			}
-
-			return event.which;
-		},
-
-		pageX: function( event ) {
-			var eventDoc, doc, body;
-
-			// Calculate pageX if missing and clientX available
-			if ( event.pageX == null && event.clientX != null ) {
-				eventDoc = event.target.ownerDocument || document;
-				doc = eventDoc.documentElement;
-				body = eventDoc.body;
-
-				return event.clientX +
-					( doc && doc.scrollLeft || body && body.scrollLeft || 0 ) -
-					( doc && doc.clientLeft || body && body.clientLeft || 0 );
-			}
-
-			return event.pageX;
-		},
+	addProp: function( name, hook ) {
```

Comment:
The existing hooks only ensure `which`/`pageX`/`pageY` properties (which this PR does automatically for all events, making reuse pointless), and aren't even defined for an event type until first dispatch: https://jsfiddle.net/1mayydrz/ . Heck, `fixHooks` is only documented at https://learn.jquery.com/events/event-extensions/#jquery-event-fixhooks-object (i.e., not at api.jquery.com). I think these changes are safe for 3.0.